monkey/common/agent_plugins/agent_plugin.py

```diff
@@ -20,4 +22,7 @@ class AgentPlugin(InfectionMonkeyBaseModel):
     # https://github.com/pydantic/pydantic/pull/1844
     config_schema: Dict[str, Any]
     default_config: Dict[str, Any]
-    source_archive: bytes
+    source_archive: B64Bytes
+
+    class Config:
+        json_encoders = {bytes: lambda byte_field: b64encode(byte_field).decode()}
```

Comment:
b64encode().decode() is confusing, but necessary. I'd recommend a comment for clarification.

```suggestion
        # b64encode() returns bytes, so we call decode() to transform bytes to str
        json_encoders = {bytes: lambda byte_field: b64encode(byte_field).decode()}
```

Reply:
I added it, but in general I don't like these types of comments. It's not the responsibility of comments to explain pythons' methods. But I agree that bytes is not what you would expect from a method named `b64encode`

Reply:
I agree. The other option is to write a new method that handles and use that instead of the lambda. The type hints will achieve what this comment achieves, more or less.